File: Programmers/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(int[] numbers) {
        /*
        String[] stringNumbers = Arrays.stream(numbers)
            .mapToObj(String::valueOf)
            .toArray(String[]::new);
        
        Arrays.sort(stringNumbers, (a, b) -> (b + a).compareTo(a + b));
        
        if (stringNumbers[0].equals("0")) return "0";
        
        return String.join("", stringNumbers);
        */
        
        String answer = Arrays.stream(numbers)
            .mapToObj(String::valueOf)
            .sorted((a, b) -> (b + a).compareTo(a + b))
            .collect(Collectors.joining());
        
        return answer.startsWith("0") ? "0" : answer;
    }
}

File: Programmers/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(vector<int> numbers) {
    vector<string> arr;
    arr.reserve(numbers.size());
    
    for (int n : numbers)
    {
        arr.push_back(to_string(n));
    }
    
    sort(arr.begin(), arr.end(),
        [](const string& a, const string& b) {
            return b + a < a + b;
        });
    
    string result;
    for (auto& s : arr) result += s;
    
    if (result[0] == '0') return "0";
    
    return result;
}